core/execution/future/market.go

```diff
@@ -1326,9 +1340,15 @@ func (m *Market) getNewPeggedPrice(order *types.Order) (*num.Uint, error) {
 	offset := num.UintZero().Mul(order.PeggedOrder.Offset, m.priceFactor)
 	if order.Side == types.SideSell {
 		price = price.AddSum(offset)
+		// this can only happen when pegged to mid, in which case we want to round to the nearest *better* tick size
+		// but this can never cross the mid by construction as the the minimum offset is 1 tick size and all prices must be
+		// whole multiples of tick size.
 		if !num.UintZero().Mod(price, m.mkt.TickSize).IsZero() {
-			price = price.Div(price.AddSum(m.mkt.TickSize), m.mkt.TickSize)
-			price = price.Mul(price, m.mkt.TickSize)
+			price.Div(price, m.mkt.TickSize)
```

Comment:
I think you can just subtract the result from `.Mod()` from the price instead of dividing and multiplying. Or expose `.DivMod()` and just use the quotient/remainder from that in a single call for the checks.